lib/install.js

```diff
@@ -191,8 +191,9 @@ function install(gyp, argv, callback) {
         var to = opts.module_path;
         var binary_module = path.join(to,opts.module_name + '.node');
 
-        opts.ignore_cache = gyp.opts['ignore-node-pre-gyp-cache'];
-        opts.skip_cache = gyp.opts['skip-node-pre-gyp-cache'];
+        var no_caching = process.env.npm_config_node_pre_gyp_cache === 'false';
```

Comment:
I'm tempted to call `npm config get node_pre_gyp_cache` (or something of that ilk; ignoring failure) if it doesn't look like it's running under npm, because otherwise `npm i` could be different to `node-pre-gyp install`, if that makes sense?